Example/Tests/Specs/Reachability/NetworkReachabilitySpec.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  NetworkReachabilitySpec.swift
+//  Tests
+//
+//  Created by Dylan Gyesbreghs on 02/01/2019.
+//  Copyright © 2019 icapps. All rights reserved.
+//
+
+import Quick
+import Nimble
+import Mockingjay
+import Connectivity
+
+@testable import Cara
+
+class NetworkReachabilitySpec: QuickSpec {
+    
+    class MockedNetworkReachability: NetworkReachability {        
+        override func startListening() {
+            guard let listener = listener else { return }
+            listener(ConnectivityStatus.connectedViaWiFi)
+        }
+    }
+    
+    override func spec() {
+        describe("CodableSerializer") {
```

Comment:
What I mostly do when testing external libraries like these. I create a mock object like you did above and I stub all the methods that could potentially be called by Cara.

ex.
```swift
class MockedNetworkReachability: NetworkReachability {
    private(set) var didTriggerStartListening: Bool = false
    override func startListening() {
        didTriggerStartListening = true
    }
}
```

Here is what I test:
```swift
it("should trigger start listening") {
    service.theFunctionThatTriggersStartListening()
    expect(networkReachability.didTriggerStartListening) == true
}
```

I would not trigger it on the mocked object though. What I would do in this case I would create a `MockedConnectivity` and test if the correct method is triggered on that class.

If it's unclear, I can show you in one of my projects somewhere this week.

Reply:
I have made a couple of changes, but it will be useful if you can show a example next week 🙂